panic: runtime error: invalid memory address or nil pointer dereference

the linter panics on simple `json.Marshal(nil)` line.
I understand that this is the bad code, that should never exists, but we have it and the linter definitely should not panic on such code

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x10f8938]

goroutine 1275 [running]:
go/ast.(*Object).Pos(0x12914a0?)
	/go/go1.20/src/go/ast/scope.go:89 +0x18
github.com/junk1tm/musttag.run.func1({0x13371e8?, 0xc005cd8880})
	/musttag/musttag.go:163 +0x17e
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0xc00444c120, {0xc000501818?, 0x1?, 0xc000501870?}, 0xc000501828)
	/go/pkg/mod/golang.org/x/tools@v0.3.0/go/ast/inspector/inspector.go:77 +0x8b
github.com/junk1tm/musttag.run(0xc008a401a0, 0xc000501a38)
	/musttag/musttag.go:135 +0x137
github.com/junk1tm/musttag.New.func1(0x12991a0?)
	/musttag/musttag.go:104 +0x33c
golang.org/x/tools/go/analysis/internal/checker.(*action).execOnce(0xc001b75180)
	/go/pkg/mod/golang.org/x/tools@v0.3.0/go/analysis/internal/checker/checker.go:791 +0xe3d
sync.(*Once).doSlow(0x100523d?, 0x1257ca0?)
	/go/go1.20/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
	/go/go1.20/src/sync/once.go:65
golang.org/x/tools/go/analysis/internal/checker.(*action).exec(0x103f085?)
	/go/pkg/mod/golang.org/x/tools@v0.3.0/go/analysis/internal/checker/checker.go:643 +0x3d
golang.org/x/tools/go/analysis/internal/checker.execAll.func1(0x0?)
	/go/pkg/mod/golang.org/x/tools@v0.3.0/go/analysis/internal/checker/checker.go:631 +0x25
created by golang.org/x/tools/go/analysis/internal/checker.execAll
	/go/pkg/mod/golang.org/x/tools@v0.3.0/go/analysis/internal/checker/checker.go:637 +0x165
```

--- a/musttag.go
+++ b/musttag.go
@@ -160,6 +160,9 @@ func run(pass *analysis.Pass, funcs map[string]Func) (any, error) {
 		initialPos := token.NoPos
 		switch arg := arg.(type) {
 		case *ast.Ident: // e.g. json.Marshal(foo)
+			if arg.Obj == nil {
+				return // e.g. json.Marshal(nil)
+			}
 			initialPos = arg.Obj.Pos()
 		case *ast.CompositeLit: // e.g. json.Marshal(struct{}{})
 			initialPos = arg.Pos()

--- a/testdata/src/tests/tests.go
+++ b/testdata/src/tests/tests.go
@@ -468,3 +468,8 @@ func nonStructArgument() {
 	custom.Marshal(0)
 	custom.Unmarshal(nil, &[]int{})
 }
+
+// test for panic with nil object issue: https://github.com/junk1tm/musttag/issues/20
+func nilObject() {
+	json.Marshal(nil)
+}

